data/src/main/java/com/moez/QKSMS/receiver/DeleteMessagesReceiver.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) 2017 Moez Bhatti <moez.bhatti@gmail.com>
+ *
+ * This file is part of QKSMS.
+ *
+ * QKSMS is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * QKSMS is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with QKSMS.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.moez.QKSMS.receiver
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import com.moez.QKSMS.interactor.DeleteMessages
+import dagger.android.AndroidInjection
+import javax.inject.Inject
+
+class DeleteMessagesReceiver : BroadcastReceiver() {
+
+    @Inject lateinit var deleteMessages: DeleteMessages
+
+    override fun onReceive(context: Context, intent: Intent) {
+        AndroidInjection.inject(this, context)
+
+        val pendingResult = goAsync()
+        val threadId = intent.getLongExtra("threadId", 0)
+        val messageIds = intent.getLongArrayExtra("messageIds")
+        deleteMessages.execute(DeleteMessages.Params(messageIds.map { it }, threadId))
```

Comment:
Could you move the lambda onto the same line, and change `.map{ it }` to something more explicit like `.toList()`?

For example
```kotlin
deleteMessages.execute(DeleteMessages.Params(messageIds.toList(), threadId)) { pendingResult.finish() }
```